clang-tools-extra/clangd/TUScheduler.h

```diff
@@ -372,6 +377,8 @@ class TUScheduler {
   // running tasks asynchronously.
   std::optional<AsyncTaskRunner> PreambleTasks;
   std::optional<AsyncTaskRunner> WorkerThreads;
+  // Manages to build module files.
+  std::optional<ModulesBuilder> ModulesManager;
```

Comment:
do we have any reasons for making this owned (and constructed) by `TUScheduler`? (sorry if i missed some previous discussions, feel free to just point me that way if this was already discussed)

i believe making this owned by clangd-lsp-server (initialized with CDBs in `onInitialize`), and passed into clangd-server as a reference (similar to CDB), and later on passing it from clangd-server to relevant components through `ParseInput` has some merit in terms of layering.

this ensures:
- you don't need to change every call-site of `buildPreamble`
- `TUScheduler` stays unchanged and doesn't need to care about any of this going forward, which is important as it's already quite complicated.
- `ParseInput`s is the natural place for these types of inputs into parse operations :)
- we can have more direct control over the `CDB` we want to provide into module-builder.
- this type of per-instance state is owned by clangd-lsp-server
- controlling lifetime deps between CDB and ModuleBuilder is more explicit